pyomo/repn/parameterized_quadratic.py

```diff
@@ -0,0 +1,439 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.common.numeric_types import native_numeric_types
+from pyomo.core.expr.numeric_expr import (
+    DivisionExpression,
+    Expr_ifExpression,
+    mutable_expression,
+    PowExpression,
+    ProductExpression,
+)
+from pyomo.repn.linear import (
+    ExitNodeDispatcher,
+    _handle_division_ANY_constant,
+    _handle_expr_if_const,
+    _handle_pow_ANY_constant,
+    _handle_product_ANY_constant,
+    _handle_product_constant_ANY,
+    _initialize_exit_node_dispatcher,
+)
+from pyomo.repn.parameterized_linear import (
+    define_exit_node_handlers as _param_linear_def_exit_node_handlers,
+    ParameterizedLinearRepnVisitor,
+    to_expression,
+    _handle_division_ANY_pseudo_constant,
+)
+from pyomo.repn.quadratic import QuadraticRepn, _mul_linear_linear
+from pyomo.repn.util import ExprType
+
+
+_FIXED = ExprType.FIXED
+_CONSTANT = ExprType.CONSTANT
+_LINEAR = ExprType.LINEAR
+_GENERAL = ExprType.GENERAL
+_QUADRATIC = ExprType.QUADRATIC
+
+
+def _merge_dict(dest_dict, mult, src_dict):
+    """
+    Slightly different from `merge_dict`
+    in the `parameterized_linear` module.
+    """
+    if not is_equal_to(mult, 1):
+        for vid, coef in src_dict.items():
+            if vid in dest_dict:
+                dest_dict[vid] += mult * coef
+            else:
+                dest_dict[vid] = mult * coef
+    else:
+        for vid, coef in src_dict.items():
+            if vid in dest_dict:
+                dest_dict[vid] += coef
+            else:
+                dest_dict[vid] = coef
```

Comment:
Despite the comment, this method is functionally identical to https://github.com/Pyomo/pyomo/blob/b5825e9e146552690d942903386102105251464c/pyomo/repn/parameterized_linear.py#L47-L59

We should probably be import / reuse it here (even through it starts with an underscore).